pkg/server/autoconfig/acprovider/provider.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package acprovider
+
+import "github.com/cockroachdb/cockroach/pkg/server/autoconfig/autoconfigpb"
+
+// Provider is an itnerface through which the auto config runner
+// can receive new tasks to run.
+type Provider interface {
```

Comment:
Something about this interface feels a little off. I guess it's the fact that it always hands you all the tasks, and that it seems to be something you tell about task completion, but it doesn't have a clear contract around how that affects behavior. The channel-based API also feels a bit fishy. 

Is it important for the client to get more than one task at a time?
What about something like:

```
// Peek will block waiting for the first task the provider believes still needs to be run.
// It will return an error if the context is canceled while waiting.
Peek(ctx context.Context) (autoconfigpb.Task, error)

// Pop will report the completion of all tasks with ID up to completed.
Pop(ctx context.Context, completed TaskID) error
```